iot-hub/Samples/device/FileUploadSample/FileUploadSample.cs

```diff
@@ -21,20 +19,24 @@ public FileUploadSample(DeviceClient deviceClient)
 
         public async Task RunSampleAsync()
         {
-            using (var fileStreamSource = new FileStream(FilePath, FileMode.Open))
-            {
-                var fileName = Path.GetFileName(fileStreamSource.Name);
+            const string filePath = "TestPayload.txt";
 
-                Console.WriteLine("Uploading File: {0}", fileName);
+            using var fileStreamSource = new FileStream(filePath, FileMode.Open);
+            var fileName = Path.GetFileName(fileStreamSource.Name);
 
-                var watch = Stopwatch.StartNew();
+            Console.WriteLine("Uploading File: {0}", fileName);
 
-                // Note: UploadToBlobAsync will use HTTPS as protocol, regardless of the DeviceClient protocol selection.
-                await _deviceClient.UploadToBlobAsync(fileName, fileStreamSource).ConfigureAwait(false);
-                watch.Stop();
+            var fileUploadTime = Stopwatch.StartNew();
 
-                Console.WriteLine("Time to upload file: {0}ms\n", watch.ElapsedMilliseconds);
-            }
+            // TODO: update this sample to use the new APIs
+
+#pragma warning disable CS0618 // Type or member is obsolete
+            // Note: UploadToBlobAsync will use HTTPS as protocol, regardless of the DeviceClient protocol selection.
+            await _deviceClient.UploadToBlobAsync(fileName, fileStreamSource);

```

Comment:
This spot will be for @timtay-microsoft

Reply:
```csharp
        public async Task RunSampleAsync()
        {
            const string filePath = "TestPayload.txt";

            using var fileStreamSource = new FileStream(filePath, FileMode.Open);
            var fileName = Path.GetFileName(fileStreamSource.Name);

            var watch = Stopwatch.StartNew();

            var fileUploadSasUriRequest = new FileUploadSasUriRequest()
            {
                BlobName = fileName
            };

            // Note: GetFileUploadSasUriAsync and CompleteFileUploadAsync will use HTTPS as protocol regardless of the DeviceClient protocol selection.
            Console.WriteLine("Getting SAS URI from IoT Hub to use when uploading the file...");
            var fileUploadSasUriResponse = await _deviceClient.GetFileUploadSasUriAsync(fileUploadSasUriRequest);

            Uri sasUri = new Uri(String.Format(
                CultureInfo.InvariantCulture,
                "https://{0}/{1}/{2}{3}",
                fileUploadSasUriResponse.HostName,
                fileUploadSasUriResponse.ContainerName,
                Uri.EscapeDataString(fileUploadSasUriResponse.BlobName), // Pass URL encoded device name and blob name to support special characters
                fileUploadSasUriResponse.SasToken));

            Console.WriteLine("Successfully got SAS URI from IoT Hub: {0}", sasUri);

            try 
            {
                Console.WriteLine("Uploading File: {0} using the Azure Storage SDK and the retrieved SAS URI for authentication", fileName);

                // Note that other versions of the Azure Storage SDK can be used here. For the latest version, see
                // https://github.com/Azure/azure-sdk-for-net/tree/master/sdk/storage#azure-storage-libraries-for-net
                CloudBlockBlob blob = new CloudBlockBlob(sasUri);
                var uploadTask = blob.UploadFromStreamAsync(fileStreamSource);
            }
            catch (Exception e) 
            {
                Console.WriteLine("Failed to upload file to Azure Storage using the Azure Storage SDK: {0}", e);

                var failedFileUploadCompletionNotification = new FileUploadCompletionNotification()
                {
                    // Mandatory. Must be the same value as the correlation id returned in the sas uri response
                    CorrelationId = fileUploadSasUriResponse.CorrelationId,

                    // Mandatory. Will be present when service client receives this file upload notification
                    IsSuccess = false,

                    // Optional, user defined status code. Will be present when service client receives this file upload notification
                    StatusCode = 500,

                    // Optional, user defined status description. Will be present when service client receives this file upload notification
                    StatusDescription = e.Message
                };

                // Note that this is done even when the file upload fails. IoT Hub has a fixed number of SAS URIs allowed active
                // at any given time. Once you are done with the file upload, you should free your SAS URI so that other
                // SAS URIs can be generated. If a SAS URI is not freed through this API, then it will free itself eventually
                // based on how long SAS URIs are configured to live on your IoT Hub.
                Console.WriteLine("Notifying IoT Hub that the file upload failed and that the SAS URI can be freed");
                await _deviceClient.CompleteFileUploadAsync(failedFileUploadCompletionNotification);
                Console.WriteLine("Notified IoT Hub that the file upload failed and that the SAS URI can be freed successfully");

                watch.Stop();
                return;
            }

            Console.WriteLine("Successfully uploaded the file to Azure Storage");

            var successfulFileUploadCompletionNotification = new FileUploadCompletionNotification()
            {
                // Mandatory. Must be the same value as the correlation id returned in the sas uri response
                CorrelationId = fileUploadSasUriResponse.CorrelationId,

                // Mandatory. Will be present when service client receives this file upload notification
                IsSuccess = true,

                // Optional, user defined status code. Will be present when service client receives this file upload notification
                StatusCode = 200,

                // Optional, user defined status description. Will be present when service client receives this file upload notification
                StatusDescription = "File upload was a success!"
            };

            Console.WriteLine("Notifying IoT Hub that the file upload succeeded and that the SAS URI can be freed");
            await _deviceClient.CompleteFileUploadAsync(successfulFileUploadCompletionNotification);
            Console.WriteLine("Notified IoT Hub that the file upload succeeded and that the SAS URI can be freed successfully");

            watch.Stop();

            Console.WriteLine("Time to upload file: {0}ms\n", watch.ElapsedMilliseconds);
        }
```

Reply:
@timtay-microsoft we're using the oldest version of the storage SDK here. Can we find one done in this millennium?